e2e/Dockerfile

```diff
@@ -2,9 +2,10 @@ FROM golang:1.19 as builder
 
 WORKDIR /go/src/github.com/Azure/aks-app-routing-operator
 ADD ../. .
-RUN go test -c -o e2e/e2e.test --count=1 --tags=e2e -c ./e2e
 
-FROM golang:1.19
-WORKDIR /go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go test -c -o e2e/e2e.test --count=1 --tags=e2e -c ./e2e
+
+FROM scratch
```

Comment:
yep! standard build from the previous PR wouldn't work with private cluster. Figured this makes more sense anyway since it's how we build the operator.